### PR TITLE
CORE-1256: Stellar fixes (from real world test)

### PR DESCRIPTION
**Pull request by carlcherry**
Fix compiler warnings as well

### CODE REVIEW

File: WalletKitCore/src/stellar/BRStellarEncode.c [hunk 1]
```diff
@@ -70,17 +70,17 @@ uint8_t * pack_int64(int64_t i, uint8_t* buffer)
...
 
-uint8_t * pack_fopaque(uint8_t *data, size_t dataSize, uint8_t *buffer)
+uint8_t * pack_fopaque(uint8_t *data, int32_t dataSize, uint8_t *buffer)
 {
     // See if the datasize is a multiple of 4 - and determine how many
     // bytes we need to pad at the end of this opaque data
-    size_t paddedSize = ((dataSize+3)/4) * 4;
-    size_t padding = paddedSize - dataSize;
+    size_t paddedSize = (((size_t)dataSize+3)/4) * 4;
+    size_t padding = paddedSize - (size_t)dataSize;
```
Comment (EBGToo) [inline WalletKitCore/src/stellar/BRStellarEncode.c:78]: Isn't `size_t` *always* the preferred size for all data sizes?  And `ssize_t` if something can be negative?

File: WalletKitCore/src/stellar/BRStellarEncode.c [hunk 2]
```diff
@@ -70,17 +70,17 @@ uint8_t * pack_int64(int64_t i, uint8_t* buffer)
...
     memcpy(buffer, data, dataSize);
     return (buffer + dataSize + padding);
 }
 
-uint8_t * pack_opaque(uint8_t *data, int dataSize, uint8_t *buffer)
+uint8_t * pack_opaque(uint8_t *data, int32_t dataSize, uint8_t *buffer)
```
Comment (EBGToo) [inline WalletKitCore/src/stellar/BRStellarEncode.c:83]: `size_t`?

**Review by EBGToo — APPROVED** [review]

File: WalletKitCore/src/stellar/BRStellarEncode.c (hunk 2 above)
Comment (carlcherry) [reply WalletKitCore/src/stellar/BRStellarEncode.c:83]: In this case no - inside of pack_opaque it first packs the size of the data (which needs to be int32_t)

File: WalletKitCore/src/stellar/BRStellarEncode.c (hunk 1 above)
Comment (carlcherry) [reply WalletKitCore/src/stellar/BRStellarEncode.c:78]: pack_fopaque is mostly called after a call to pack_int which takes an int32_t.  It is seldom called directly.

**Pull request merged**